Update home screen

karolinkas opened the pull request:
## Description

This implements the redesign of the ColonyHome page in combination with the addition of two new simple core components: an expandable paragraph and a breadcrumb component. 

- [x] New ColonyMeta layout has been created
- [x] Domains list is on the left of the page
- [x] Tabs component has been shuffled about
- [x] Domain breadcrumb has been added to top of page
- [x] extract ExtendParagrpah component
- [x]  added available funds in sidebar which gets wired in #1783

## TODO
- [x] edit .md files for styleguide
- [x] move more button a tiny bit
- [x] extract BreadCrumb completely

## New UI Screenshot
<img width="1274" alt="Screen Shot 2019-09-12 at 15 23 19" src="https://user-images.githubusercontent.com/6294044/64787625-48264880-d571-11e9-8475-5ec0ebbf9a43.png">


## ExpandableParagraph component in action
![expandableParagraph](https://user-images.githubusercontent.com/6294044/64786725-5ecba000-d56f-11e9-9ec3-707f9b1b2fd0.gif)

## BreadCrumb component in action
![breadCrumb](https://user-images.githubusercontent.com/6294044/64786910-df8a9c00-d56f-11e9-852c-a2a77ec8779d.gif)


Resolves #1785

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyHome.css:
```diff
@@ -3,19 +3,37 @@
   composes: stretchVertical stretchHorizontal from '~styles/layout.css';
   display: grid;
-  padding: 130px 80px; /* vertical padding of 60px + nav height (70px) */
+  padding: 16px 18px; /* vertical padding of 60px + nav height (70px) */
```
The comment values should be changed as well, right ?

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyHome.tsx:
```diff
@@ -220,21 +240,73 @@ const ColonyHome = ({
+        }
+        return accumulator;
+      },
+      ['Root'],
```
There's two things here:
- This should be a message descriptor _(But this might be a little hard, please look into it)_
- As everywhere else, the root domain should be lowercase, ie: `root`

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyMeta/ColonySubscribe.css:
```diff
@@ -1,13 +1,13 @@
-  top: -10px;
-  left: -26px;
+  top: 19px;
+  left: -11px;
```
This should have a comment as to why it's a negative value

karolinkas commented on src/modules/dashboard/components/ColonyHome/ColonyDomains/ColonyDomains.tsx:
```diff
@@ -35,6 +36,7 @@ const ColonyDomains = ({
   colonyAddress,
   setFilteredDomainId,
   filteredDomainId,
+  noTitle,
```
Can do, so far there's only one occurrence, so it's hard to say what's better.

karolinkas commented:
@rdig I applied all changes you mentioned above, please check one more time.

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyHome.tsx:
```diff
@@ -220,21 +240,79 @@ const ColonyHome = ({
+          />
+        </div>
+        <div className={styles.domainContainer}>
+          <ColonyDomains
```
I just noticed this:

`ColonyDomains` need to be refactored to be part of the `ColonyMeta` component, not external to it

karolinkas replied:
@rdig Could you tell me a reason or an argument to cause more work?

karolinkas replied:
Also I just realised a little icon I missed in the DomainsList I assume, it's to collapse the list, so it will have it's own state, saying that I would say it's better to be in it's own component. 
<img width="75" alt="Screen Shot 2019-09-19 at 14 18 02" src="https://user-images.githubusercontent.com/6294044/65243342-757e7380-dae8-11e9-94fa-c64610a41f6a.png">

rdig replied:
> @rdig Could you tell me a reason or an argument that justifies this, because it is causing more work.

Because both from a logical point of view, and from a code point of view they're part of the meta information regarding a colony.

From a code point of view, after they're moved into the sidebar component (aka: Colony Meta), you actually simply a bunch of styles, as they can now be applied globally (you won't need to individually set margins and padding on sections)

> because it is causing more work.

If you don't want to do it here, I'll do it in my downstream branch #1839, but either way this is happening

rdig replied:
> Also I just realised a little icon I missed in the DomainsList I assume, it's to collapse the list, so it will have it's own state, saying that I would say it's better to be in it's own component.

I'm not 100% sure that is to collapse / show the list. It might be there to add new domains strait from the new Colony Home.

Better ask @JoinColony/product to make sure

Karol-colony replied:
please ignore this icon

karolinkas replied:
> If you don't want to do it here, I'll do it in my downstream branch #1839, but either way this is happening

I'll do it just wanted to understand the reasoning behind.

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyMeta/ColonyMeta.css:
```diff
@@ -13,10 +13,15 @@
+  border-bottom: 1px solid var(--temp-grey-5);
 }
 
 .dynamicSection {
```
The `.dynamicSection` class is not used in `ColonyMeta`

Please clean up the styles file

rdig commented on src/modules/dashboard/components/ColonyHome/ColonyHome.tsx:
```diff
@@ -220,21 +240,79 @@ const ColonyHome = ({
+    .sort((a, b) => a.id - b.id)
+    .reduce(
+      (accumulator, element) => {
+        if (element.id <= filteredDomainId && element.name) {
```
I think this is breaking the production builds, based on what Circle is telling us: https://app.circleci.com/jobs/github/JoinColony/colonyDapp/20579

Please check the production build manually locally

karolinkas commented:
Ok, so I have double-triple check everything and weirdly enough I couldn't reproduce some things in Chrome:
- the empty states of the taskList are showing well, independent of the filter option
![Screen Shot 2019-09-19 at 17 25 42](https://user-images.githubusercontent.com/6294044/65258049-8b009700-db02-11e9-80a3-fcac1f51db41.png)

- the dividers are perfectly well aligned for me
![Screen Shot 2019-09-19 at 17 26 18](https://user-images.githubusercontent.com/6294044/65258114-9eabfd80-db02-11e9-900e-4da59561967d.png)

Everything else is adjusted and ready!

rdig commented:
> the dividers are perfectly well aligned for me

@ceolson01 if you some time, can you please check how the separator alignment is rendering for you ?

Me and Karolin get very different results.

ceolson01 commented:
Firstly, let me say this is looking awesome @karolinkas and @JoinColony/product 👏 

### Alignment

I was able to reproduce the divider alignment issue with the viewport width at ~1125px and below, but it *is* aligned at greater widths. But the alignment does seem fragile / coincidental. I think we need to play with the `grid-row` property a bit, or maybe even adjust how these elements are grouped.

Here's a demo:

![Divider Alignment](https://user-images.githubusercontent.com/3052635/65342482-78d13800-db98-11e9-9295-33305e782656.gif)

### More / Hide Link

I also found an interesting little bug 🐛  - the `More` / `Hide` link seems to jump around a little bit for me.

**Short text demo:**

![More Jump](https://user-images.githubusercontent.com/3052635/65343291-40caf480-db9a-11e9-8e94-c1526d5bfdf1.gif)

**Long text demo:**

![More Jump Long](https://user-images.githubusercontent.com/3052635/65343303-47f20280-db9a-11e9-8580-41557f02d636.gif)

rdig commented:
@karolinkas please have a look / fix the issues found by Curtis, then I'll re-review this.

karolinkas commented:
@rdig I could reproduce the button jumping bug and fixed it but not the alignment issue with small screen sizes, but made some adjustments that should fix it. Please check if it's fixed for you too.

karolinkas commented:
@rdig Ok, I simplified ColonyHome the annoying by removing that complicated MessageDescriptor creation and adjusted the outstanding UI issues. This should be ready to merge now.

Karol-colony commented:
"Currently, it displays all domains as a sub-domain of the previous one, which clearly is wrong as they are all top-level domains (under root, of course)" right, as you said. 

We don't have intention to display nested domains at the current stage. We are not ready with implications of this one. I guess, there wouldn't be so many intends, only: core domain / nested domain; It would be good if also someone else from @JoinColony/product could confirm that.

karolinkas commented:
@rdig I get your point of not being able to display nesting with the current breadcrumb component, but also currently it's not possible to change the way they are being displayed since  all share the same parentId. Maybe it goes out of the scope of this PR?

rdig commented:
> since all share the same parentId

But if they all have the same parent id, shouldn't they all have the same 1-level deep nesting ? That's what I'm asking for.

karolinkas commented:
@rdig Yes, completely, but it seems impossible to visualize that with a breadCrumb component or do you have a specific idea?

rdig commented:
All domains that you select should basically render like the first one _("Public Ballot" in this case)_

![Screenshot from 2019-09-26 15-01-09](https://user-images.githubusercontent.com/1193222/65687553-3add8380-e072-11e9-9924-49c5a29c6166.png)

So no matter what you select, since they're all under root should show:
```
Root > Domain Name
```

itspatch commented:
I think it is fine if we implement this the way Raul has suggested and see how it goes. It is the most "correct" to have Root > Domain Name. 

Originally, it was designed so that it would just say All Domains > Domain Name. It didn't matter if that Domain Name was Root or any other domain. So the breadcrumb would be All Domains > Root or All Domains > Public Ballot. This isn't technically correct because Public Ballot is not a subdomain of All Domains, but with this breadcrumb would allow the user to get back to where they came from (All Domains). 

However, I like educating the user about where they are in the domain hierarchy, so I am ok with going with Root > Domain Name.

karolinkas commented:
@rdig Thanks for explaining, that was actually easy to implement. I commited it.
